app/models/user.rb

```diff
@@ -103,8 +103,35 @@ def decline_invite
     update(invited_by_id: nil)
   end
 
-  private
-    def email_without_domain
-      email.split('@').first
+  def process_unit(unit_params, user_is_guest)
+    if user_is_guest # if user is guest
+      if unit = self.unit # and has a unit
```

Comment:
Indeed I am checking an assignment here. Line 108 is saying "If we succeed in assigning self.unit to unit" (true), then do some things with that unit including destroy it, then create and assign a new unit. It is also implicitly saying "if we don't succeed in assigning self.unit to unit" (false), then just create and assign a new unit.